shell/browser/api/electron_api_browser_window.cc

```diff
@@ -92,7 +93,7 @@ BrowserWindow::BrowserWindow(gin::Arguments* args,
   InitWithArgs(args);
 
   // Install the content view after BaseWindow's JS code is initialized.
-  SetContentView(gin::CreateHandle<View>(isolate, web_contents_view.get()));
+  window()->GetContentsView()->AddChildViewAt(web_contents_view->view(), 0);
```

Comment:
Before:
- Window
  - ContentView (WebContentsView)

After:
- Window
  - ContentView
    - WebContentsView

Is this the expected hierarchy of the views? My understanding is that WebContentsView is a child which has me confused by this line:
> This PR has BrowserWindow's WebContentsView be a **sibling** to the content view

Reply:
No, your comment here is inaccurate—"GetContentsView()" here is actually not doing what you think it's doing :)

Reply:
Ahhh, I missed the 's' in `GetContentsView()`. If my understanding is correct, a more accurate picture might look something like this now:

- Window
  - ContentsView (root)
    - WebContentsView
    - ContentView
      - BrowserView (A)
      - BrowserView (B)
      - ...

I think for now this is fine and keeps the expected behavior. In the future, it'd be great if we can simplify it to:

- Window
  - ContentsView (root)
    - ContentView
      - WebContentsView
      - ~~BrowserView~~ WebContentsView
      - ...

Maybe that's possible already by using BaseWindow directly.

Reply:
I don't think the hierarchy you proposed is desirable for BrowserWindow, as BrowserWindow has a lot of functionality that depends on having one single "blessed" WebContentsView. Having the "blessed" WebContentsView be part of ContentView means it would be accessible to JS, which could e.g. remove it, leading to strange behavior.